benchmark/single-source/StringReplaceSubrange.swift

```diff
@@ -0,0 +1,116 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+let tags: [BenchmarkCategory] = [.validation, .api, .String]
+
+public let StringReplaceSubrange = [
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.String",
+    runFunction: { replaceSubrange($0, "coffee", with: "t") },
+    tags: tags
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.LargeManaged.String",
+    runFunction: { replaceSubrange($0, largeManagedString, with: "t") },
+    tags: tags,
+    setUpFunction: setupLargeManagedString
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.Substr",
+    runFunction: { replaceSubrange($0, "coffee", with: getSubstring("t")) },
+    tags: tags
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.LargeManaged.Substr",
+    runFunction: { replaceSubrange($0, largeManagedString, with: getSubstring("t")) },
+    tags: tags,
+    setUpFunction: setupLargeManagedString
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.ArrChar",
+    runFunction: { replaceSubrange($0, "coffee", with: getArrayCharacter(Array<Character>(["t"]))) },
+    tags: tags
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.LargeManaged.ArrChar",
+    runFunction: { replaceSubrange($0, largeManagedString, with: getArrayCharacter(Array<Character>(["t"]))) },
+    tags: tags,
+    setUpFunction: setupLargeManagedString
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.SmallLiteral.RepeatedChar",
+    runFunction: { replaceSubrange($0, "coffee", with: getRepeatedCharacter(repeatedCharacter)) },
+    tags: tags
+  ),
+  BenchmarkInfo(
+    name: "Str.replaceSubrange.LargeManaged.RepeatedChar",
+    runFunction: { replaceSubrange($0, largeManagedString, with: getRepeatedCharacter(repeatedCharacter)) },
+    tags: tags,
+    setUpFunction: setupLargeManagedString
+  ),
+]
+
+// MARK: - Privates for String
+
+private var largeManagedString: String = {
+    return getString("coffee\u{301}coffeecoffeecoffeecoffee")
+}()
+
+private func setupLargeManagedString() {
+    _ = largeManagedString
+}
+
+// MARK: - Privates for Repeated<Character>
+
+private let repeatedCharacter: Repeated<Character> = {
+  let character = Character("c")
+  return repeatElement(character, count: 1)
+}()
+
+
+@inline(never)
+private func replaceSubrange(_ N: Int, _ string: String, with replacingString: String) {
+  var copy = getString(string)
+  let range = string.startIndex..<string.index(after: string.startIndex)
+  for _ in 0 ..< 5_000 * N {
+    copy.replaceSubrange(range, with: replacingString)
+  }
```

Comment:
What is the criteria for choosing this multiplying number like 5000? Does this depend on the benchmark time?

Reply:
Correct. We are just trying to size the workload to run in 20–1000 μs, so that it is in a measurement sweet spot for our system.

Reply:
Thanks for explaining!